### ISSUE TITLE
ETH2 fetch price problem

### ISSUE THREAD

**Issue by Larision** [eprbell/dali-rp2#191]
i dont know why not resolve all times good price...

```
2023-06-07 23:23:57,410/Coinbase/fran/DEBUG: Transaction: {"id": "xxxxxxxxxxxxxxxxxxxxxx", "type": "staking_reward", "status": "completed", "amount": {"amount": "0.00000314", "currency": "ETH2"}, "native_amount": {"amount": "0.00", "currency": "EUR"}, "description": null, "created_at": "2023-03-30T20:33:28Z", "updated_at": "2023-03-30T20:33:28Z", "resource": "transaction", "resource_path": "/v2/accounts/xxxxxxxxxxxxxxxxxxxxxx/transactions/xxxxxxxxxxxxxxxxxxxxxx", "instant_exchange": false, "from": {"id": "xxxxxxxxxxxxxxxxxxxxxx", "resource": "user", "resource_path": "/v2/users/xxxxxxxxxxxxxxxxxxxxxx", "currency": "ETH2"}, "details": {"title": "Recompensa de Ethereum 2", "subtitle": "De Coinbase", "header": "Se han recibido 0,00000314 ETH2 (0,00 \u20ac)", "health": "positive"}, "hide_native_amount": false}
2023-06-07 23:23:57,410/Coinbase/fran/DEBUG: Transaction: {"id": "xxxxxxxxxxxxxxxxxxxxxx", "type": "staking_reward", "status": "completed", "amount": {"amount": "0.00000322", "currency": "ETH2"}, "native_amount": {"amount": "0.01", "currency": "EUR"}, "description": null, "created_at": "2023-03-22T08:38:32Z", "updated_at": "2023-04-27T14:58:36Z", "resource": "transaction", "resource_path": "/v2/accounts/xxxxxxxxxxxxxxxxxxxxxx/transactions/xxxxxxxxxxxxxxxxxxxxxx", "instant_exchange": false, "from": {"id": "xxxxxxxxxxxxxxxxxxxxxx", "resource": "user", "resource_path": "/v2/users/xxxxxxxxxxxxxxxxxxxxxx", "currency": "ETH2"}, "details": {"title": "Recompensa de Ethereum 2", "subtitle": "De Coinbase", "header": "Se han recibido 0,00000322 ETH2 (0,01 \u20ac)", "health": "positive"}, "hide_native_amount": false}
2023-06-07 23:23:57,410/Coinbase/fran/DEBUG: Transaction: {"id": "xxxxxxxxxxxxxxxxxxxxxx", "type": "staking_reward", "status": "completed", "amount": {"amount": "0.00000626", "currency": "ETH2"}, "native_amount": {"amount": "0.01", "currency": "EUR"}, "description": null, "created_at": "2023-03-17T23:02:07Z", "updated_at": "2023-04-27T14:58:36Z", "resource": "transaction", "resource_path": "/v2/accounts/xxxxxxxxxxxxxxxxxxxxxx/transactions/xxxxxxxxxxxxxxxxxxxxxx", "instant_exchange": false, "from": {"id": "xxxxxxxxxxxxxxxxxxxxxx", "resource": "user", "resource_path": "/v2/users/xxxxxxxxxxxxxxxxxxxxxx", "currency": "ETH2"}, "details": {"title": "Recompensa de Ethereum 2", "subtitle": "De Coinbase", "header": "Se han recibido 0,00000626 ETH2 (0,01 \u20ac)", "health": "positive"}, "hide_native_amount": false}
2023-06-07 23:23:57,410/Coinbase/fran/DEBUG: Transaction: {"id": "xxxxxxxxxxxxxxxxxxxxxx", "type": "staking_reward", "status": "completed", "amount": {"amount": "0.00000298", "currency": "ETH2"}, "native_amount": {"amount": "0.01", "currency": "EUR"}, "description": null, "created_at": "2023-03-14T22:28:42Z", "updated_at": "2023-04-27T14:58:36Z", "resource": "transaction", "resource_path": "/v2/accounts/xxxxxxxxxxxxxxxxxxxxxx/transactions/xxxxxxxxxxxxxxxxxxxxxx", "instant_exchange": false, "from": {"id": "xxxxxxxxxxxxxxxxxxxxxx", "resource": "user", "resource_path": "/v2/users/xxxxxxxxxxxxxxxxxxxxxx", "currency": "ETH2"}, "details": {"title": "Recompensa de Ethereum 2", "subtitle": "De Coinbase", "header": "Se han recibido 0,00000298 ETH2 (0,01 \u20ac)", "health": "positive"}, "hide_native_amount": false}

```

```
2023-06-07 23:27:43,816/dali/DEBUG: Self-contained transaction: InTransaction:
  plugin=Coinbase
  unique_id=xxxxxxxxxxxxxxxxxxxxxx
  raw_data={"id": "xxxxxxxxxxxxxxxxxxxxxxxx", "type": "staking_reward", "status": "completed", "amount": {"amount": "0.00000314", "currency": "ETH2"}, "native_amount": {"amount": "0.00", "currency": "EUR"}, "description": null, "created_at": "2023-03-30T20:33:28Z", "updated_at": "2023-03-30T20:33:28Z", "resource": "transaction", "resource_path": "/v2/accounts/xxxxxxxxxxxxxxxxxxxxxx/transactions/xxxxxxxxxxxxxxxxxxxxxx", "instant_exchange": false, "from": {"id": "xxxxxxxxxxxxxxxxxxxxxx", "resource": "user", "resource_path": "/v2/users/xxxxxxxxxxxxxxxxxxxxxx", "currency": "ETH2"}, "details": {"title": "Recompensa de Ethereum 2", "subtitle": "De Coinbase", "header": "Se han recibido 0,00000314 ETH2 (0,00 \u20ac)", "health": "positive"}, "hide_native_amount": false}
  timestamp=2023-03-30 20:33:28+0000
  asset=ETH2
  exchange=Coinbase
  holder=fran
  transaction_type=Staking
  spot_price=1639.11765252
  crypto_in=0.00000314
  fiat_fee=0
  fiat_in_no_fee=None
  fiat_in_with_fee=None
  notes=nearest spot_price read from CCXT-converter plugin; Recompensa de Ethereum 2
2023-06-07 23:27:43,816/dali/DEBUG: Self-contained transaction: InTransaction:
  plugin=Coinbase
  unique_id=a8ff0a86-e793-59b9-8a11-63184dad2482
  raw_data={"id": "xxxxxxxxxxxxxxxxxxxxxx", "type": "staking_reward", "status": "completed", "amount": {"amount": "0.00000322", "currency": "ETH2"}, "native_amount": {"amount": "0.01", "currency": "EUR"}, "description": null, "created_at": "2023-03-22T08:38:32Z", "updated_at": "2023-04-27T14:58:36Z", "resource": "transaction", "resource_path": "/v2/accounts/xxxxxxxxxxxxxxxxxxxxxx/transactions/xxxxxxxxxxxxxxxxxxxxxx", "instant_exchange": false, "from": {"id": "xxxxxxxxxxxxxxxxxxxxxx", "resource": "user", "resource_path": "/v2/users/xxxxxxxxxxxxxxxxxxxxxx", "currency": "ETH2"}, "details": {"title": "Recompensa de Ethereum 2", "subtitle": "De Coinbase", "header": "Se han recibido 0,00000322 ETH2 (0,01 \u20ac)", "health": "positive"}, "hide_native_amount": false}
  timestamp=2023-03-22 08:38:32+0000
  asset=ETH2
  exchange=Coinbase
  holder=fran
  transaction_type=Staking
  spot_price=3105.590062111801242236024845
  crypto_in=0.00000322
  fiat_fee=0
  fiat_in_no_fee=0.01
  fiat_in_with_fee=0.01
  notes=Recompensa de Ethereum 2
2023-06-07 23:27:43,816/dali/DEBUG: Self-contained transaction: InTransaction:
  plugin=Coinbase
  unique_id=xxxxxxxxxxxxxxxxxxxxxx
  raw_data={"id": "xxxxxxxxxxxxxxxxxxxxxx", "type": "staking_reward", "status": "completed", "amount": {"amount": "0.00000626", "currency": "ETH2"}, "native_amount": {"amount": "0.01", "currency": "EUR"}, "description": null, "created_at": "2023-03-17T23:02:07Z", "updated_at": "2023-04-27T14:58:36Z", "resource": "transaction", "resource_path": "/v2/accounts/xxxxxxxxxxxxxxxxxxxxxx/transactions/xxxxxxxxxxxxxxxxxxxxxx", "instant_exchange": false, "from": {"id": "xxxxxxxxxxxxxxxxxxxxxx", "resource": "user", "resource_path": "/v2/users/xxxxxxxxxxxxxxxxxxxxxx", "currency": "ETH2"}, "details": {"title": "Recompensa de Ethereum 2", "subtitle": "De Coinbase", "header": "Se han recibido 0,00000626 ETH2 (0,01 \u20ac)", "health": "positive"}, "hide_native_amount": false}
  timestamp=2023-03-17 23:02:07+0000
  asset=ETH2
  exchange=Coinbase
  holder=fran
  transaction_type=Staking
  spot_price=1597.444089456869009584664537
  crypto_in=0.00000626
  fiat_fee=0
  fiat_in_no_fee=0.01
  fiat_in_with_fee=0.01
  notes=Recompensa de Ethereum 2
2023-06-07 23:27:43,816/dali/DEBUG: Self-contained transaction: InTransaction:
  plugin=Coinbase
  unique_id=xxxxxxxxxxxxxxxxxxxxxx
  raw_data={"id": "xxxxxxxxxxxxxxxxxxxxxx", "type": "staking_reward", "status": "completed", "amount": {"amount": "0.00000298", "currency": "ETH2"}, "native_amount": {"amount": "0.01", "currency": "EUR"}, "description": null, "created_at": "2023-03-14T22:28:42Z", "updated_at": "2023-04-27T14:58:36Z", "resource": "transaction", "resource_path": "/v2/accounts/xxxxxxxxxxxxxxxxxxxxxx/transactions/xxxxxxxxxxxxxxxxxxxxxx", "instant_exchange": false, "from": {"id": "xxxxxxxxxxxxxxxxxxxxxx", "resource": "user", "resource_path": "/v2/users/xxxxxxxxxxxxxxxxxxxxxx", "currency": "ETH2"}, "details": {"title": "Recompensa de Ethereum 2", "subtitle": "De Coinbase", "header": "Se han recibido 0,00000298 ETH2 (0,01 \u20ac)", "health": "positive"}, "hide_native_amount": false}
  timestamp=2023-03-14 22:28:42+0000
  asset=ETH2
  exchange=Coinbase
  holder=fran
  transaction_type=Staking
  spot_price=3355.704697986577181208053691
  crypto_in=0.00000298
  fiat_fee=0
  fiat_in_no_fee=0.01
  fiat_in_with_fee=0.01
  notes=Recompensa de Ethereum 2
```
see spot price of transactions... 3355?

**Comment by macanudo527** [comment]
Have you tried using the historic_crypto pair converter since ETH2 appears to be Coinbase-only asset.

**Comment by macanudo527** [comment]
What exactly is it? It doesn't appear to be publicly traded. I get that it is staked ETH, but do you trade it on Coinbase or is it locked? Can you convert it to EUR at any time?

**Comment by Larision** [comment]
right, eth2 is eth staked, was created at merge fork. coinbase change it to ETH at any time.

**Comment by macanudo527** [comment]
And it is convertable 1 to 1? 1 ETH2 for 1 ETH at anytime?

I think this should be resolved with an alias. #140 

It might take 1-2 months before we can get those working. I'm almost finished with a major upgrade optimizing the CCXT graph, after that I'll get started on it after that.

**Comment by Larision** [comment]
i think only called ETH2 when in stacking. unstack convert it in ETH automatically.
My wallet shows just ETH. but getting rewards in ETH2 its crazy......
![image](https://github.com/eprbell/dali-rp2/assets/41712973/308f9d66-6cc4-4ca2-b204-f4e21da04cb1)
and still getting ETH2 rewards....

```
2023-05-31T17:27:06Z,Rewards   Income,ETH2,2.89e-06,EUR,1748.76,0.010000,0.010000,0.00,"Received   0,00000289 ETH2 from Coinbase Rewards"
2023-06-03T20:38:32Z,Rewards   Income,ETH2,3.16e-06,EUR,1764.05,0.010000,0.010000,0.00,"Received   0,00000316 ETH2 from Coinbase Rewards"
2023-06-06T18:05:31Z,Rewards   Income,ETH2,3.13e-06,EUR,1755.29,0.010000,0.010000,0.00,"Received   0,00000313 ETH2 from Coinbase Rewards"
2023-06-10T23:19:07Z,Rewards   Income,ETH2,3.67e-06,EUR,1631.97,0.010000,0.010000,0.00,"Received   0,00000367 ETH2 from Coinbase Rewards"
2023-06-15T16:52:25Z,Rewards   Income,ETH2,2.61e-06,EUR,1496.96,0.00,0.00,0.00,"Received 0,00000261   ETH2 from Coinbase Rewards"
```

**Comment by macanudo527** [comment]
This should been resolved with #196 . If it didn't, please open another issue to address it.
